src/sentry/api/serializers/models/team.py

```diff
@@ -166,9 +167,9 @@ def get_attrs(
             project_map[project_team.team_id].append(projects_by_id[project_team.project_id])
 
         external_teams_map = defaultdict(list)
-        for external_team in external_teams:
-            serialized = serialize(external_team, user)
-            external_teams_map[external_team.team_id].append(serialized)
+        serialized_list = serialize(external_actors, user, key="team")
+        for serialized in serialized_list:
+            external_teams_map[serialized["teamId"]].append(serialized)
```

Comment:
Could use 
`for external_actor, serialized in zip(external_actors, serialized_list)`

Then you still have access to the original `team_id`, rather than potentially stringified id.